src/components/generics/vacation-test-forms/my-vacation-requests.tsx

```diff
@@ -65,6 +71,70 @@ const RenderVacationRequests = () => {
     initializeRequests();
   }, [person]);
 
+  /**
+   * Initializes all vacation request statuses
+   */
+  const initializeRequestStatuses = async () => {
+    if (!person) return;
+
+    try {
+      const vacationRequestStatuses: VacationRequestStatus[] = [];
+
+      const statusesApi = Api.getVacationRequestStatusApi(accessToken?.access_token);
+
+      await Promise.all(requests.map(async request => {
+        const createdStatuses = await statusesApi.listVacationRequestStatuses({ id: request.id! });
+        createdStatuses.forEach(createdStatus => {
+          vacationRequestStatuses.push(createdStatus);
+        });
+      }));
+
+      setStatuses(vacationRequestStatuses);
+    } catch (error) {
+      context.setError(strings.errorHandling.fetchVacationDataFailed, error);
+    }
+  };
+
+  useEffect(() => {
+    if (requests.length <= 0) {
```

Comment:
Again this condition can be shortened to (!requests.length)

Reply:
shortened it to !requests.length